src/main/java/net/snowflake/client/core/auth/ClientAuthnDTO.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All right reserved.
+ */
+package net.snowflake.client.core.auth;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+
+@SnowflakeJdbcInternalApi
+public class ClientAuthnDTO {
+
+  // contains all the required data for current authn step
+  public final Map<String, Object> data;
```

Comment:
Maybe we can make it private by the way?

Reply:
It's being serialized by Jackson, so I guess we cannot

Reply:
But we can add getters/setters ;)

Reply:
I'd prefer not to add them if we are not planning to add any additional logic

Reply:
As discussed, I'll add a getter.